test/e2e/federated-namespace.go

```diff
@@ -55,9 +54,20 @@ var _ = framework.KubeDescribe("Federation namespace [Feature:Federation]", func
 
 		AfterEach(func() {
 			framework.SkipUnlessFederated(f.Client)
-			deleteAllTestNamespacesFed14(f.FederationClientset_1_4)
+			// TODO: set wait to true once NS controller is fixed.
+			deleteAllTestNamespaces(
+				f.FederationClientset_1_4.Core().Namespaces().List,
+				func(ns *api_v1.Namespace) error {
```

Comment:
How about also passing f.FederationClientset_1.4.Core().Namespaces().Delete

here (and below)?